feat: improve filter by users

SjoenH opened the pull request:
New design for filtering processes by users

closes #229

dansvend commented on pages/processes/index.tsx:
```diff
@@ -1,36 +1,44 @@
 
   const router = useRouter();
-
+  const requiredUsers = router.query?.user
+    ?.toString()
+    .split(",")
+    .map((user) => parseInt(user));
+  const requiredLabels = router.query?.rl
+    ?.toString()
+    .split(",")
+    .map((label) => parseInt(label));
```
Consider creating creating one/two helper function(s) and reuse this on the three pages

dansvend commented on pages/processes/favourite.tsx:
```diff
@@ -1,32 +1,42 @@
 
   const router = useRouter();
-
+  const requiredUsers = router.query?.user
+    ?.toString()
+    .split(",")
+    .map((user) => parseInt(user));
+  const requiredLabels = router.query?.rl
+    ?.toString()
+    .split(",")
+    .map((label) => parseInt(label));
```
Consider creating creating one/two helper function(s) and reuse this on the three pages

dansvend reviewed (approved):
Clean code👌 I added a couple of questions

SjoenH commented on components/FilterUserButton.tsx:
```diff
@@ -0,0 +1,35 @@
+        }}
+        onClick={() => setVisibleScrim(true)}
+      >
+        Filter by users{" "}
```
Good suggestion. Will fix this

SjoenH replied:
Fixed

SjoenH commented on components/Labels/FilterLabelButton.tsx:
```diff
@@ -1,9 +1,15 @@
 import { Button, Scrim } from "@equinor/eds-core-react";
 import React, { useState } from "react";
+
```
I'm automatically sorting my imports. And I guess it likes to separate them. Will take a look.

dansvend replied:
If that's standard functionality for a code formatter I'm all for having it like you have it

SjoenH commented on components/Labels/FilterLabelButton.tsx:
```diff
@@ -16,7 +22,8 @@ export default function FilterLabelButton(): JSX.Element {
         onClick={() => setVisibleScrim(true)}
       >
-        Filter by label
+        Filter by label{" "}
```
Will do

SjoenH commented on pages/processes/favourite.tsx:
```diff
@@ -1,32 +1,35 @@
         items: itemsPerPage,
         onlyFavorites: true,
-        ...router.query,
+        ru: stringToArray(router.query.user),
+        rl: stringToArray(router.query.rl),
```
I believe this is cleaner. Let me know what you think :) @dansvend

dansvend replied:
Perfect👌

dansvend reviewed (approved):
Clean👌

SjoenH commented:
Changed the text on no results to: “No processes match your search criteria.”
Added the active-filter-section to my-page and favourites-page.
Fixed crash when you use your own user as a filter while on my-page.
Fixed the search-field.
Not sure why the font is misbehaving… I'll fix it later when upgrading some packages
Upgrade everything #354

dansvend reviewed (approved):
Looks good💪
